src/transaction/client.rs

```diff
@@ -24,12 +24,15 @@ impl Client {
     /// use tikv_client::{Config, TransactionClient};
     /// use futures::prelude::*;
     /// # futures::executor::block_on(async {
-    /// let connect = TransactionClient::connect(Config::default());
-    /// let client = connect.await.unwrap();
+    /// let client = TransactionClient::new(Config::default()).await.unwrap();
     /// # });
     /// ```
-    pub fn connect(config: Config) -> Connect {
-        Connect::new(config)
+    pub async fn new(config: Config) -> Result<Client> {
+        let bg_worker = ThreadPool::new()?;
```

Comment:
This is fine for now, but long term we will want to let the user pass in a thread pool (or callback) of some kind.

Reply:
Exactly. But there isn't a uniform abstraction of a threadpool in the async ecosystem. Futures has Spawn trait, tokio has Executor trait...